### PR TITLE
Explain both persistence and permanence in relation to IPFS. #364

### PR DESCRIPTION
**Pull request by alexmmueller**
@johnnymatthews This is the first draft. Any feedback is welcome! I am wondering if it makes sense to link to the Privacy in IPFS" section as it is surely related to the concepts here. What do you think? The  other tasks:

- [x] Briefly explain how IPFS handles data (this is already touched on by referring to IPFS as a cache in the current version of this page).
- [x] Explain what Permanence is, and how IPFS enables the Permanent Web
- [ ] Explain the relationship (or not) between Permanence and Persistence
- [x] Explain what pinning is.
- [x] Explain why pinning is important if a user wants to keep files available
- [x] Describe what pinning services are, and popular services currently available.
- [x] Link to the how-to/pin-files guide.

have beed addressed (or were addressed already).

### CODE REVIEW

**Comment by johnnymatthews** [comment]
Thanks for this PR @alexmmueller! I'll review it shortly.

File: docs/concepts/persistence.md [hunk 1]
```diff
@@ -6,17 +6,23 @@ description: Learn about how IPFS treats persistence and permanence on the web,
...
+
+Nodes on the IPFS network cache resources that are downloaded through them and make those resources available for upload to other nodes. This system depends on nodes being willing and able to cache and share resources with the network. Storage isn’t infinite, however, so nodes need to clear out some of their previously cached resources to make room for new resources. In order to ensure that data stays available on IPFS (aka is persisted) and mustn't be thrown away data can be "pinned" to one or more IPFS nodes. "Pinning" a CID tells an IPFS node that the data is important and mustn't be thrown away.
 
 You should pin any content you consider important in order to ensure that content is retained over the long term. Since data important to someone else may not be important to you, pinning enables you to have control over the disk space and data retention you need.
```
Comment (AlexxNica) [inline docs/concepts/persistence.md:17]:
```suggestion
In simple terms, pinning gives you control over disk space and data retention. As such, you should use that control to pin any content you wish to keep on IPFS indefinitely.
```

Try to maintain these explanation sentences as short as possible. Simple and common words also help make a concise sentence, causing a hard concept to be easier to understand.

File: docs/concepts/persistence.md [hunk 2]
```diff
@@ -6,17 +6,23 @@ description: Learn about how IPFS treats persistence and permanence on the web,
...
 
 ## Pinning in context
 
-Your IPFS node can store data based on different kinds of user events. For instance, you can add a file with `ipfs add ...`. It will also store data you request, such as by loading a web page through the gateway (`http://localhost:8080/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco`) or with `ipfs cat ...`. Your node will consult with other IPFS peers to find these requested data, and will store the results in the local cache. `ipfs add` will automatically pin the content, but other IPFS commands do not include automatic pinning.
+::: tip
+If you're interested in how pinning files fits into the overall lifecycle of data in IPFS, check out this video from IPFS Camp 2019! [Core Course: The Lifecycle of Data in Dweb](https://www.youtube.com/watch?v=fLUq0RkiTBA)
```
Comment (AlexxNica) [inline docs/concepts/persistence.md:22]:
```suggestion
If you want to learn more about how pinning fits into the overall lifecycle of data in IPFS, check out the course from the IPFS Camp 2019: [Core Course: The Lifecycle of Data in DWeb](https://www.youtube.com/watch?v=fLUq0RkiTBA)
```

Improve formatting and rewording for a more outgoing invitation.

File: docs/concepts/persistence.md [hunk 3]
```diff
@@ -6,17 +6,23 @@ description: Learn about how IPFS treats persistence and permanence on the web,
...
+:::
+
+Your IPFS node can store data based on different kinds of user events. For instance, you can add a file with `ipfs add ...`. It will also store data you request, such as by loading a web page through the gateway (`http://localhost:8080/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco`) or with `ipfs cat ...`. Your node will consult with other IPFS peers to find these requested data, and will store the results in the local cache. `ipfs add` will automatically pin the content, but other IPFS commands do not include automatic pinning. More information on file pinning options can be found in [the pinning guide](https://docs.ipfs.io/how-to/pin-files/#three-kinds-of-pins)
```
Comment (AlexxNica) [inline docs/concepts/persistence.md:25]:
```suggestion
Your IPFS node can store data based on different kinds of user events. For example, you can add a file using the CLI command [`ipfs add`](https://docs.ipfs.io/reference/cli/#ipfs-add). It also automatically stores data you request (e.g., by loading a web page through the gateway) or with [`ipfs cat`](https://docs.ipfs.io/reference/cli/#ipfs-cat). Your node will check if any peers have the requested data, and if so, will store the results in the local cache.
```

Improve clarity and add references to CLI commands for easy follow-up.

File: docs/concepts/persistence.md [hunk 4]
```diff
@@ -6,17 +6,23 @@ description: Learn about how IPFS treats persistence and permanence on the web,
...
 
 When garbage collection is triggered on a node, any pinned content is automatically exempt from deletion. Non-pinned data may be deleted; if you request it again later, the data can be retrieved from another node.
```
Comment (AlexxNica) [inline docs/concepts/persistence.md:27]:
```suggestion
As a reminder, not every CLI command will automatically pin content, but specific ones do, as is the case of `ipfs add.`
```

Shorten the sentence and improve readability.

File: docs/concepts/persistence.md [hunk 5]
```diff
@@ -27,3 +33,11 @@ To ensure that your important data is retained, you may want to use a pinning se
 - You don’t have a lot of disk space, but you want to ensure some data sticks around.
 - Your computer is a laptop, phone, or tablet that will have intermittent connectivity to the network, but you want to be able to access your data on IPFS from anywhere at any time, even when the device you added it from is offline.
 - You want a backup that ensures your data is always available from another computer on the network in case you accidentally delete or garbage-collect on your own computer.
```
Comment (AlexxNica) [inline docs/concepts/persistence.md:35]:
```suggestion
- You don't have a lot of disk space but want to ensure some data sticks around.
- You have a device with poor network connectivity but still want your data on IPFS to be available anywhere, at any time.
- You want a readily available backup for your data to protect it from accidental deletions or garbage collection.
```

Fix apostrophe punctuation mark and improve readability.

**Comment by AlexxNica** [comment]
Hey there, @alexmmueller! I've added some suggestions to your work that I think will help it be even better than it already is!

I hope it helps, and they're not by any means "official" (I'm not from the IPFS staff) or required. ✌️

**Review by AlexxNica — COMMENTED** [review]
Hey there, @alexmmueller! I've added some suggestions to your work that I think will help it be even better than it already is!

I hope it helps, and they're not by any means "official" (I'm not from the IPFS staff) or required. ✌️

File: docs/concepts/persistence.md [hunk 6]
```diff
@@ -6,17 +6,23 @@ description: Learn about how IPFS treats persistence and permanence on the web,
 
 # Persistence, permanence and pinning
```
[the remark this answers is not in the document]
Comment (alexmmueller) [reply docs/concepts/persistence.md:7]: Thank you for your great improvement suggestions @AlexxNica

**Comment by alexmmueller** [comment]
I resolved a last conflict and included proposed changes from @AlexxNica.

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
I've made a couple of edits and added some suggestions. The main thing I think we're missing here is a section on garbage-collection and how it works within IPFS.

File: docs/concepts/persistence.md [hunk 7]
```diff
@@ -6,13 +6,27 @@ description: Learn about how IPFS treats persistence and permanence on the web a
...
+
+## Persistence versus permanence
+
+One of the main goals of IPFS is to preserve humanity's history by enabling the permanent web. But what does permanence mean? And why does this matter?
+
+Today [The average lifespan of a web page is 100 days](https://blogs.loc.gov/thesignal/2011/11/the-average-lifespan-of-a-webpage/) before it's gone forever. It's not good enough for the primary medium of our era to be this fragile. Therefore IPFS keeps every version of your files and makes it simple to set up resilient networks for mirroring data.
+
+Trying to counter that, nodes on the IPFS network automatically cache the downloaded resources and keep those available for uploading to other nodes. This system depends on nodes being willing and able to cache and share resources with the network. Storage is finite, so nodes need to clear out some of their previously cached resources to make room for new resources. To ensure that data stays available on IPFS (i.e., persisted) and not be thrown away, they can be "pinned" to one or more IPFS nodes. When "pinning," a CID tells an IPFS node that the data is important and it shouldn't throw it away.
```
[the remark this answers is not in the document]
Comment (alexmmueller) [reply docs/concepts/persistence.md:17]: Thank you @johnnymatthews. Sounds good. I will make the requested changes and add a section on garbage collection.

[the remark this answers is not in the document]
Comment (alexmmueller) [reply docs/concepts/persistence.md]: @johnnymatthews I added a section on garbage collection and included your requested changes. Let me know ahat you think. Thanks!

File: docs/concepts/persistence.md [hunk 8]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
+
+## Garbage Collection
+
+[Garbage collection](<https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)>) is a form of automatic resource management widely used in software applications. Garbage collection reliefs programmers and users from manually allocating or deallocating finite resources such as memory or disk storage. IPFS uses garbage collection to free disk space on ypur IPFS node as needed.
+
+The IPFS garbage collector (or GC for short) is configured in the `Datastore`section of [the go-ipfs config file](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md). The important settings related to the garbage colllector are:
+
+- `StorageGCWatermark` The percentage of the StorageMax value at which a garbage collection will be triggered automatically if the daemon was run with automatic gc enabled (that option defaults to false currently). Default: 90
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md:27]:
```suggestion
- `StorageGCWatermark`: The percentage of the `StorageMax` value at which a garbage collection will be triggered automatically, if the daemon is running with automatic garbage collection enabled. The default is 90`.
```

File: docs/concepts/persistence.md [hunk 9]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
+
+- `GCPeriod` A time duration specifying how frequently to run a garbage collection. Only used if automatic gc is enabled. Default: 1h
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md:29]:
```suggestion
- `GCPeriod`: Specify how frequently garbage collection should run. Only used if automatic garbage collection is enabled. The default is 1 hour.
```

File: docs/concepts/persistence.md [hunk 10]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
+
+As described in `StorageGCWatermark`, currently garbage collection does not run automatically. To run it manually run [`ipfs repo gc`(https://docs.ipfs.io/reference/cli/#ipfs-repo-gc).
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md]:
```suggestion
To manually start garbage collection, [run `ipfs repo gc`](https://docs.ipfs.io/reference/cli/#ipfs-repo-gc):

```bash
ipfs repo gc

> removed QmPZhyTu8D7NqR5NvgkgNYsSYD4CNjnyuFejB8i23itJvA
> removed QmSYQFVAZgEnpa6NxiW5agyj3XU9VR4CbERShXiLhuPPPE
> removed QmS6SJXApoi59hqD8Naktgakc6UNHK1XDhqhtMg9sBhY8g
```
```

File: docs/concepts/persistence.md [hunk 11]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
+
+To enable automatic garbage collection you have to run [ìpfs daemon](https://docs.ipfs.io/reference/cli/#ipfs-daemon) command to start your IPFS node with the `--enable-gc`option.
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md]:
```suggestion
To enable automatic garbage collection use `--enable-gc` when starting the IPFS daemon:

```bash
ipfs daemon --enable-gc

> Initializing daemon...
> go-ipfs version: 0.7.0
> Repo version: 10
> ...
```
```

File: docs/concepts/persistence.md [hunk 12]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
+
+::: tip
+If you use IPFS Desktop or the IPFS Web UI the settings related to garbage collection can be found under `settings`. You can also directly run the garbage collector by clicking on the taskbar icon of the IPFS Desktop application and selecting `Àdvanced` > `Run Garbage Collector`.
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md:36]:
```suggestion
If you use IPFS Desktop or the IPFS Web UI the settings related to garbage collection can be found in the **Settings** tab. You can also directly run the garbage collector by clicking on the taskbar icon of the IPFS Desktop application and selecting **Advanced** → **Run Garbage Collector**.
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
Added a couple of improvements, but apart from them, this PR looks good!

File: docs/concepts/persistence.md (hunk 11 above)
Comment (alexmmueller) [reply docs/concepts/persistence.md]: If you use IPFS Desktop or the IPFS Web UI the settings related to garbage collection can be found in the **Settings** tab. You can also directly run the garbage collector by clicking on the taskbar icon of the IPFS Desktop application and selecting **Advanced** → **Run Garbage Collector**.

File: docs/concepts/persistence.md (hunk 12 above)
Comment (alexmmueller) [reply docs/concepts/persistence.md:36]: Changed as suggested. Thank you!

File: docs/concepts/persistence.md (hunk 10 above)
Comment (alexmmueller) [reply docs/concepts/persistence.md]: Changed as suggested. Thank you!

File: docs/concepts/persistence.md (hunk 9 above)
Comment (alexmmueller) [reply docs/concepts/persistence.md:29]: Changed as suggested. Thank you!

File: docs/concepts/persistence.md [hunk 13]
```diff
@@ -14,9 +14,27 @@ One of the main goals of IPFS is to preserve humanity's history by enabling the
...
 
-In simple terms, pinning gives you control over disk space and data retention. As such, you should use that control to pin any content you wish to keep on IPFS indefinitely.
+To ensure that data _persists_ on IPFS, and is not deleted during garbage collection, [data can be pinned](/how-to/pin-files/) to one or more IPFS nodes. In simple terms, pinning gives you control over disk space and data retention. As such, you should use that control to pin any content you wish to keep on IPFS indefinitely.
+
+## Garbage Collection
+
+[Garbage collection](<https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)>) is a form of automatic resource management widely used in software applications. Garbage collection reliefs programmers and users from manually allocating or deallocating finite resources such as memory or disk storage. IPFS uses garbage collection to free disk space on ypur IPFS node as needed.
+
+The IPFS garbage collector (or GC for short) is configured in the `Datastore`section of [the go-ipfs config file](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md). The important settings related to the garbage colllector are:
```
[the remark this answers is not in the document]
Comment (alexmmueller) [reply docs/concepts/persistence.md:25]: Changed as suggested. Thank you!

**Comment by alexmmueller** [comment]
Included the requested changes and fixed typo. @johnnymatthews Let me know if we can improve the PR further. Thank you!

File: docs/concepts/persistence.md [hunk 14]
```diff
@@ -24,8 +73,14 @@ To ensure that your important data is retained, you may want to use a pinning se
 
 Some available pinning service providers are:
 
+- You don't have a lot of disk space but want to ensure some data sticks around.
+- You have a device with poor network connectivity but still want your data on IPFS to be available anywhere, at any time.
+- You want a readily available backup for your data to protect it from accidental deletions or garbage collection.
```
Comment (johnnymatthews) [inline docs/concepts/persistence.md:78]:
```suggestion
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]

**Review by johnnymatthews — APPROVED** [review]

**Comment by alexmmueller** [comment]
@johnnymatthews Is there anything left to be done in order to gain the bounty?

**Pull request merged**